### ISSUE TITLE
Add Yale fonts to code base

### ISSUE THREAD

**Issue by FCRodriguez7** [yalelibrary/YUL-DC#626]
rra  12:42 PM @mikeapp fyi https://static-uat.yalepages.org and https://static.yalepages.org all set now for fonts****

**Acceptance Criteria**

- [x] Add previously described fonts 
- [x] Ensure fonts from the Yale Roman Suit is added
   - [x] YaleDisplay-Roman
   - [x] YaleNew-Bold 
   - [x] YaleNew-BoldItalic
   - [x] YaleNew-Italic 
   - [x] YaleNew-Roman
   - [x] MalloryMP Medium
   - [x] MalloryMP Black
- [x] Update codebase to point at yale pages
- [x] Support CORS headers as required

**Comment by FCRodriguez7** [comment]
**Acceptance Criteria**

- [ ] Add previously described fonts 
- [ ] Ensure fonts from the Yale Roman Suit is added
   - [ ] YaleDisRom
   - [ ] YaleNew-Bold 
   - [ ] YaleNew-BoldItalic
   - [ ] YaleNew-Italic 
   - [ ] YaleNew-Roman

**Comment by alishaevn** [comment]
All I see at https://static.yalepages.org/ is a screen that says "static.yalepages.org". I see that the fonts are loaded in the yul-dc-static-assets repo, but don't know how to access them with a url.

@mikeapp would you be able to provide me with the url to one of the fonts from that repo that I can use as an example? 

![image.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/bb9e90a5-9a5a-44ef-a3b1-b38b78bcc9cb)

**Comment by alishaevn** [comment]
talked about this in stand up and the path from the yul-dc-static-assets repo is the path we should use for the fonts too. the link below downloads a font so that works!

https://static.yalepages.org/fonts/yaledisplay/YaleNew-Bold.otf

**Comment by alishaevn** [comment]
I've added the 4 yale and 8 mallory fonts that are in the static repo to the codebase. however, there are 2 fonts: Mallory-MPBlack and Mallory MP-Medium (not the same as Mallory-Medium) that are in the zeplin design, but not in the static repo.

mike stated that he added all Mallory fonts that he had to the repo so I've reached out to Matt via email to see where Mike can find the two "MP" fonts so they can be added and used as well.

**Comment by mikeapp** [comment]
It turns out that I missed these, they were in a parallel directory in the download.  I'll add them tonight and then ask DevOps to redeploy (if necessary).

**Comment by alishaevn** [comment]
no response from Matt yet.

**Comment by alishaevn** [comment]
mallory-medium on the test library site
![image.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/34047e4a-be55-490d-8e5a-14f7ac6daa14)

mallory-medium locally with the new url
![image.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/e4efb0d9-f192-4fff-9566-9552afb29000)

**Comment by mikeapp** [comment]
Download site:  https://yaleidentity.yale.edu/typefaces 

I did not include the Postscript font:

Yale Display (Type 1 Postscript)
“Yale Display” is an adaptation of the Yale typeface designed to optimize its appearance when set at large sizes. It features sharp serifs, high contrast between thick and thin strokes, tall x-height, and tight letterspacing. It is useful for web and print display of all kinds, generally in sizes greater than 24 points. While Yale Display is included in the Yale typeface download package, it is currently available only as a Postscript Type 1 font, compatible only with Macintosh operating systems.

**Comment by alishaevn** [comment]
Still no reply from Matt yet about where to find the mallory mp fonts.

during standup yesterday it was discussed to load various formats of each font for better cross browser compatibility. currently all of the yale fonts are `.otf` and all of the mallory fonts are `.woff`.

I converted all of the yale display and yale new fonts into `.eot`, `.svg`, `.ttf` and `.woff` and emailed those to mike since Yale owns that font. However, permission would have to be granted by the makers of the Mallory font to convert those as well. The version of Mallory that I found on [dafont free](https://www.dafontfree.io/mallory-font-family/) said that it's for personal use only. the version on [frere jones](https://frerejones.com/families/mallory) is for purchase though. I'm not sure how this font was acquired by yale to know what rights they have.

I've looped @mikeapp into the email thread with matt because he has questions about the rights too.

![Screen Shot 2020-10-01 at 12.47.31 PM.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/2d38afd4-2f1d-4fd8-a75a-a36ae05e0d3e)

**Comment by mikeapp** [comment]
1) Mallory is definitely licensed and we can't get other fonts except through Yale's identity/printer's office
2) I'm not sure what the licensing is for the "Yale" font, and whether it precludes changing formats. 

I will reach out to Matt.

**Comment by dl-maura** [comment]
Waiting on redeploy from DevOps

**Comment by mikeapp** [comment]
@alishaevn  DevOps did the redeploy, YaleDisplay should be available.

**Comment by alishaevn** [comment]
all "mallory", "yale new" and "yale display" fonts have been updated on my feature branch.

waiting to hear back from matt about where to find the "mallory mp" fonts on the email thread between him, mike and myself.

**Comment by dl-maura** [comment]
we have the mallory, will add

**Comment by alishaevn** [comment]
Waiting on the Mallory MP fonts to be added to the yale repo

**Comment by alishaevn** [comment]
# MR
https://github.com/yalelibrary/yul-dc-blacklight/pull/325

# Expected behavior
the following fonts should now be sourced from https://static.yalepages.org:
##### multiple font formats
- YaleDisplay-Roman
- YaleDesign-SmallCaps
- YaleMarks
- YaleNew-Bold
- YaleNew-BoldItalic
- YaleNew-Italic
- YaleNew-Roman

##### .woff format only due to rights restrictions
- Mallory-Bold
- Mallory-BoldItalic
- Mallory-Book
- Mallory-BookItalic
- Mallory-Light
- Mallory-LightItalic
- Mallory-Medium
- Mallory-MediumItalic

# Caveat
the two fonts below are still missing from the fonts repo. mike has them, they just have to be uploaded. don't want to allow them to be a blocker for the rest of the fonts though so I'm setting this as ready for review and will create a new PR when the two fonts below get uploaded.
- MalloryMP Medium
- MalloryMP Black

**Comment by ra359** [comment]
Please use the following URLs (CNAMEs to the yalepages.org sites), The certs will work fine there:

https://static.library.yale.edu/
https://static-uat.library.yale.edu/
https://static-test.library.yale.edu/

**Comment by alishaevn** [comment]
in conversation about the above with mike and rick

**Comment by mikeapp** [comment]
Added MalloryMP Medium and Black fonts to test/uat/production branches, will submit request to update on servers.

**Comment by mikeapp** [comment]
Redeploy completed.

**Comment by alishaevn** [comment]
investigating CORS errors with @mikeapp 
![image.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/2bf22915-32da-4a1b-a6a0-42397cd72d16)

**Comment by mikeapp** [comment]
So this is currently a SpinUp S3 bucket, which doesn't allow us to make changes to the CORS settings as far as I can tell.  I've put in ticket.  If we move it to a new S3 bucket in our account I can set up CORS no problem.   DNS might take some time.

**Comment by ra359** [comment]
Tenyo over at ITS says we _should_ be able to do this from the s3 cli, i'm looking into this now...

**Comment by ra359** [comment]
Mike A. - can you validate that CORS is enabled correctly on https://static-test.library.yale.edu .

**Comment by mikeapp** [comment]
No go.  The traffic is being routed through Cloudfront, I think that introduces additional problems: `x-cache: Miss from cloudfront`
We should put this on hold and set up a static site under our control when we have time.

**Comment by ra359** [comment]
Can you try one more time before we throw in towel? We just made another config change.

**Comment by mikeapp** [comment]
@ra359  Headers look good!  Can you push this out to uat and production?   All of the servers are looking to production for their fonts, so that will be the real test.

**Comment by ra359** [comment]
Should be all set now. As instructed, used the following on tst/uat

<CORSConfiguration>
 <CORSRule>
   <AllowedOrigin>*</AllowedOrigin>
   <AllowedHeader>*</AllowedHeader>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedMethod>PUT</AllowedMethod>
   <AllowedMethod>POST</AllowedMethod>
   <AllowedMethod>DELETE</AllowedMethod>
   <MaxAgeSeconds>3000</MaxAgeSeconds>
 </CORSRule>
</CORSConfiguration>

and the following on PRD:

<CORSConfiguration>
 <CORSRule>
   <AllowedOrigin>*.library.yale.edu</AllowedOrigin>
   <AllowedHeader>*</AllowedHeader>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedMethod>PUT</AllowedMethod>
   <AllowedMethod>POST</AllowedMethod>
   <AllowedMethod>DELETE</AllowedMethod>
   <MaxAgeSeconds>3000</MaxAgeSeconds>
 </CORSRule>
</CORSConfiguration>

**Comment by mikeapp** [comment]
I see CORS headers returned from static site when testing using `curl`

Note that the production site is set to send CORS headers to *.library.yale.edu. The test and UAT sites will send the correct headers to any host, but are limited to on-campus access by other means.   So you may need to test locally using static-test.

**Comment by mikeapp** [comment]
Testing with the (temporarily using the `static-uat` host) looks good, it should work when deployed to the clusters and pointing to `static`.

**Comment by mikeapp** [comment]
Asking Rick whether we can update `static` CORS config to `*`

**Comment by mikeapp** [comment]
Rick has opened CORS on production.  I tested the branch locally and Chrome was able to load the fonts, no CORS error.

**Comment by alishaevn** [comment]
Also verified that this is now working, **while connected to the VPN**
![image.png](https://images.zenhubusercontent.com/5e5ea9bffa78052c1cbcc74f/a506e33b-0207-41e8-92a5-d8d8a03caaba)
